### PR TITLE
taxonomy diff/data generate: support taxonomy-base=empty

### PR DESCRIPTION
**Pull request by markmc**
This allows users to specify that they want all taxonomy files in their repo to be included, where 'null' symbolically represents the empty tree before this repo was created.

See also instructlab/sdg#242

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/data/generate.py
Comment (bbrowning) [inline src/instructlab/data/generate.py:68]: nit: there's an extra period at the end of this help message.

**Review by bbrowning — APPROVED** [review]
This looks reasonable to me. I don't love having to update this logic in multiple places, but that's a separate issue to tackle outside of just adding support for a null taxonomy-base here. Giving this my symbolic approval, but it will need approval from a maintainer of this repo to actually merge.

File: src/instructlab/utils.py
Comment (danmcp) [inline src/instructlab/utils.py:587]: Nit: updated may not be the best term here anymore

**Comment by n1hility** [comment]
LGTM

**Review by n1hility — APPROVED** [review]

**Review by danmcp — APPROVED** [review]

**Review by jaideepr97 — APPROVED** [review]
ran a quick test with this and it worked great
thanks @markmc

File: README.md
Comment (jaideepr97) [inline README.md:510]: nit: do we need the quotes around "empty"?

Comment (jaideepr97) [reply README.md:510]: this might break if users try to copy paste this command as is 
```
(venv) ➜  instructlab git:(ad84ac9) ilab taxonomy diff --taxonomy-base=`empty`
zsh: command not found: empty
compositional_skills/STEM/math/propositional_logic/arguments/qna.yaml
Taxonomy in /Users/jrao/Library/Application Support/instructlab/taxonomy is valid :)
```

**Comment by jaideepr97** [comment]
@Mergifyio backport release-v0.18

**Pull request merged**
